bzt/modules/jmeter.py

```diff
@@ -1184,22 +1231,29 @@ def _get_http_defaults(default_domain_name, default_port, timeout, retrieve_reso
         cfg = etree.Element("ConfigTestElement", guiclass="HttpDefaultsGui",
                             testclass="ConfigTestElement", testname="Defaults")
 
+        if retrieve_resources:
+            cfg.append(JMX._bool_prop("HTTPSampler.image_parser", True))
+            cfg.append(JMX._bool_prop("HTTPSampler.concurrentDwn", True))
+            if concurrent_pool_size:
+                cfg.append(JMX._string_prop("HTTPSampler.concurrentPool", concurrent_pool_size))
+
         params = etree.Element("elementProp",
                                name="HTTPsampler.Arguments",
                                elementType="Arguments",
                                guiclass="HTTPArgumentsPanel",
                                testclass="Arguments", testname="user_defined")
         cfg.append(params)
-        if retrieve_resources:
-            cfg.append(JMX._bool_prop("HTTPSampler.image_parser", True))
-            cfg.append(JMX._bool_prop("HTTPSampler.concurrentDwn", True))
-            if concurrent_pool_size:
-                cfg.append(JMX._string_prop("HTTPSampler.concurrentPool", concurrent_pool_size))
+        if default_url:
+            parsed_url = urlsplit(default_url)
+            if parsed_url.scheme:
+                cfg.append(JMX._string_prop("HTTPSampler.protocol", parsed_url.scheme))
+            if parsed_url.hostname:
+                cfg.append(JMX._string_prop("HTTPSampler.domain", parsed_url.hostname))
+            if parsed_url.port:
+                cfg.append(JMX._string_prop("HTTPSampler.port", parsed_url.port))
+            if parsed_url.path:
+                cfg.append(JMX._string_prop("HTTPSampler.path", parsed_url.path))
```

Comment:
This is not what we discussed. Setting path is not a value. We're not setting base URL. We're setting default protocol, hostname and port.